File: app/backend/src/main/java/com/guidematch/booking/BookingStatus.java
package com.guidematch.booking;

/**
 * 예약 상태.
 *  REQUESTED  : 여행자가 요청함 (가이드 응답 대기)
 *  ACCEPTED   : 가이드가 수락 (계약 성립)
 *  REJECTED   : 가이드가 거절
 *  CANCELLED  : 여행자가 취소
 *  COMPLETED  : 일정 종료 (이후 리뷰 가능)
 */
public enum BookingStatus {
    REQUESTED,
    ACCEPTED,
    REJECTED,
    CANCELLED,
    COMPLETED
}

File: app/frontend/src/lib/bookingStatus.ts
// 예약 상태 배지 스타일/아이콘 — 단일 소스.
// traveler/bookings, guide/requests, bookings/[id] 세 페이지가 공유한다.

export const STATUS_CLS: Record<string, string> = {
  REQUESTED: "badge-amber",
  ACCEPTED: "badge-emerald",
  REJECTED: "badge-red",
  CANCELLED: "badge-gray",
  COMPLETED: "badge-indigo",
};

export const STATUS_ICON: Record<string, string> = {
  REQUESTED: "⏳",
  ACCEPTED: "✅",
  REJECTED: "❌",
  CANCELLED: "🚫",
  COMPLETED: "🏁",
};
